dev_guide/secrets.adoc

```diff
@@ -167,6 +169,12 @@ internal DNS names), by using the CA bundle in the
 *_/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt_* file that is
 automatically mounted in their pod.
 
+The signature algorithm for this feature is `x509.SHA256WithRSA`. This feature
+is intended to support complex middleware applications that need out-of-the-box
```

Comment:
> This feature is intended to support complex middleware applications that need out-of-the-box certificates. It has the same settings as the server certificates generated by the administrator tooling for nodes and masters.

These sentences could potentially go up into the first paragraph, either before or after the first sentence there? Feels a little strange to be explaining the use case at the very end, after having gotten into the technical weeds.